examples/parallel_environment.py

```diff
@@ -0,0 +1,282 @@
+            for observations, dones in zip(batched_observations, batched_dones):
+                actions = {
+                    agent_id: agents[agent_id].act(agent_obs)
+                    for agent_id, agent_obs in observations.items()
+                    if not dones.get(agent_id, False)
+                }
+                batched_actions.append(actions)
+
+            # Step all environments in parallel
+            (
+                batched_observations,
+                batched_rewards,
+                batched_dones,
+                batched_infos,
+            ) = env.step(batched_actions)
+
+            # Sum the scores
+            for dones, infos in zip(batched_dones, batched_infos):
+                for agent_id, val in infos.items():
+                    if dones[agent_id]:
+                        tot_scores[agent_id] += val["score"]
+
+    # Print average episode score of each agent
+    ave_scores = {
+        agent_id: score / (num_episodes * num_env)
+        for agent_id, score in tot_scores.items()
+    }
+    print("Average episode score:")
+    print(f"{ave_scores}")
+
+    env.close()
+
+
+if __name__ == "__main__":
+    parser = default_argument_parser("parallel-environment-example")
+    parser.add_argument(
+        "--num-agents",
+        default=2,
+        type=int,
+        help="Number of ego agents to simulate in each environment.",
+    )
+    parser.add_argument(
+        "--num-stack",
+        default=2,
+        type=int,
+        help="Number of consecutive frames to stack in each environment's observation.",
+    )
+    parser.add_argument(
+        "--num-env",
+        default=2,
+        type=int,
+        help="Number of parallel environments to simulate.",
+    )
+    parser.add_argument(
+        "--max-episode-steps",
+        default=128,
+        type=int,
+        help="Maximum number of steps per episode.",
+    )
+    parser.add_argument(
+        "--num-steps",
+        default=1280,
+        type=int,
+        help="Total number of steps to simulate per environment in parallel asynchronous simulation.",
+    )
+    args = parser.parse_args()
+
+    print("\nParallel environments with asynchronous episodes.\n")
+    main(
+        scenarios=args.scenarios,
+        sim_name=args.sim_name,
+        headless=args.headless,
+        seed=args.seed,
+        num_agents=args.num_agents,
+        num_stack=args.num_stack,
+        num_env=args.num_env,
+        auto_reset=True,
+        max_episode_steps=args.max_episode_steps,
+        num_steps=args.num_steps,
+    )
+
+    print("\nParallel environments with synchronous episodes.\n")
+    main(
+        scenarios=args.scenarios,
+        sim_name=args.sim_name,
+        headless=args.headless,
+        seed=args.seed,
+        num_agents=args.num_agents,
+        num_stack=args.num_stack,
+        num_env=args.num_env,
+        auto_reset=False,
+        max_episode_steps=args.max_episode_steps,
+        num_episodes=args.episodes,
+    )
```

Comment:
I think we should not repeat this and just add a cli option for `auto_reset` instead.